pkg/controller/federatedcluster/controller.go

```diff
@@ -78,15 +76,19 @@ type ClusterController struct {
 }
 
 // StartClusterController starts a new cluster controller.
-func StartClusterController(config *util.ControllerConfig, stopChan <-chan struct{}, clusterMonitorPeriod time.Duration) {
+func StartClusterController(config *util.ControllerConfig, stopChan <-chan struct{}, clusterMonitorPeriod time.Duration) error {
 	fedClient, kubeClient, crClient := config.AllClients("cluster-controller")
-	controller := newClusterController(fedClient, kubeClient, crClient, config.FederationNamespaces, clusterMonitorPeriod)
+	controller, err := newClusterController(config, fedClient, kubeClient, crClient, clusterMonitorPeriod)
+	if err != nil {
+		return err
+	}
 	glog.Infof("Starting cluster controller")
 	controller.Run(stopChan)
+	return nil
 }
 
 // newClusterController returns a new cluster controller
-func newClusterController(fedClient fedclientset.Interface, kubeClient kubeclientset.Interface, crClient crclientset.Interface, namespaces util.FederationNamespaces, clusterMonitorPeriod time.Duration) *ClusterController {
+func newClusterController(config *util.ControllerConfig, fedClient client.Client, kubeClient kubeclientset.Interface, crClient crclientset.Interface, clusterMonitorPeriod time.Duration) (*ClusterController, error) {
```

Comment:
(No action required) The generic client could replace not just usage of the generated fed client, but also the kube and cr clients.  Not saying it has to happen in this PR, mind you.

Also, `fedClient` suggests a federation-specific client, but `client.Client` can be used to access any api resource, so an eventual rename is suggested.

Reply:
> The generic client could replace not just usage of the generated fed client, but also the kube and cr clients. Not saying it has to happen in this PR, mind you.

yes, it should be in subsequent pr's. Already this pr is big.

> Also, `fedClient` suggests a federation-specific client, but `client.Client` can be used to access any api resource, so an eventual rename is suggested.

Renamed the variable properly, keeping future in mind.